### PR TITLE
Split CI Jobs

### LINKED ISSUE #176: Split the build CI job into smaller parts
At some point we would like to split the build CI job into smaller parts. The link to the discussion is [here](https://github.com/P5-wrapper/react/pull/174/files#r929623740)

### PR DESCRIPTION
**Pull request by jamesrweb**
Fixes #176

## Proposed Changes

- Split composite job into individual jobs
- Utilise [the new setup job](https://github.com/P5-wrapper/setup-action) which each other job depends upon

## Additional Notes (optional)

N / A

### CODE REVIEW

File: .github/workflows/CD.yml [hunk 1]
```diff
@@ -7,20 +7,20 @@ on:
 concurrency: cd-${{ github.ref }}
 
 jobs:
-  build-and-deploy:
+  setup:
```
Comment (yevdyko) [inline .github/workflows/CD.yml]: We use the same `setup` in CI and CD workflows. What about using action composition to reduce duplication? Is this possible here?

https://github.blog/changelog/2021-08-25-github-actions-reduce-duplication-with-action-composition/

Comment (jamesrweb) [reply .github/workflows/CD.yml]: The current setup I've made doesn't even work, it seems not likely to either. Once I get things working somehow, then I will take a look at that, thanks for the tip! Seems interesting😇

Comment (jamesrweb) [reply .github/workflows/CD.yml]: Created a shared job for both places 🚀!

**Comment by jamesrweb** [comment]
@yevdyko working! 🚀

Please review the two open PRs as soon as you get time. 🎁

**Comment by jamesrweb** [comment]
@yevdyko I moved the base because the other PR is bugging out waiting for the build task that is required in settings now and so it needs the new flows to correct itself 👍🏻

**Comment by jamesrweb** [comment]
The PR automerged due to repo settings 🤷🏻‍♂️... didn't expect that actually... but it's okay since the target was the other PR anyway 😄

**Pull request merged**